met-api/src/met_api/services/report_setting_service.py

```diff
@@ -0,0 +1,145 @@
+            return True
+
+        return False
+
+    @classmethod
+    def _check_for_survey_type_component(cls, survey_id, component, user_id):
+        # Check if the component type is SURVEY then loop through each question to extract the survey questions
+        if component['type'] == FormIoComponentType.SURVEY.value:
+            questions = component['questions']
+            if not questions:
+                return
+
+            for question in questions:
+                cls._create_or_update_data_for_survey_type(survey_id, component, user_id, question)
+        else:
+            cls._create_or_update_data(survey_id, component, user_id)
+
+    @staticmethod
+    def _create_or_update_data(survey_id, component, user_id) -> ReportSettingModel:
+        report_setting_exists = ReportSettingModel.find_by_question_key(survey_id, component['key'])
+
+        # Update the record if its existing
+        if report_setting_exists:
+            report_setting_exists.question_id = component['id']
+            report_setting_exists.question = component['label']
+            report_setting_exists.updated_date = datetime.utcnow()
+            report_setting_exists.save()
+        else:
+            # Create the record if its not existing
+            report_setting = ReportSettingModel(survey_id=survey_id,
+                                                question_id=component['id'],
+                                                question_key=component['key'],
+                                                question_type=component['type'],
+                                                question=component['label'],
+                                                display=True,
+                                                created_by=user_id,
+                                                created_date=datetime.utcnow()
+                                                )
+            report_setting.save()
+
+    @staticmethod
+    def _create_or_update_data_for_survey_type(survey_id, component, user_id, question) -> ReportSettingModel:
+        # For component type SURVEY the unique identifier is a combination of key and value. The key for each
+        # question will be same as its part of a single component within form json
+        report_setting_exists = ReportSettingModel.find_by_question_key(survey_id,
+                                                                        component['key'] + '-' + question['value'])
+
+        # Update the record if its existing
+        if report_setting_exists:
+            report_setting_exists.question_id = component['id'] + '-' + question['value']
+            report_setting_exists.question_key = component['key'] + '-' + question['value']
+            report_setting_exists.question = component['label']
+            report_setting_exists.updated_date = datetime.utcnow()
+            report_setting_exists.save()
```

Comment:
this can be commit i think
Or move the report_setting_exists.save() to outside the loop